src/util/httpProvider.js

```diff
@@ -72,6 +82,73 @@ export default class HttpProvider {
     return result;
   }
 
+  static timeoutPromise(timer) {
+    return new Promise(_resolve => {
+      const ids = setTimeout(() => {
+        clearTimeout(ids);
+        // eslint-disable-next-line prefer-promise-reject-errors
+        _resolve({ type: 'timeout' });
+      }, timer);
```

Comment:
Change timer -> delay
We use timer to declare setTimtout/setInterval like
const timer = setTimeout(function() {} , delay)